logging/reformatLogger.go

```diff
@@ -0,0 +1,173 @@
+package logging
+
+import (
+	"bytes"
+	"io"
+	"github.com/go-kit/kit/log"
+	"time"
+	"fmt"
+	"strings"
+	"errors"
+	"github.com/ttacon/chalk"
+	"github.com/spf13/cast"
+)
+
+type TextFormatter struct {
+	// Force disabling colors.
+	DisableColors bool `json:"disable_colors"`
+
+	// Disables the truncation of the level text to 4 characters.
+	DisableLevelTruncation bool `json:"disable_level_truncation"`
+
+	// Whether the logger's out is to a terminal
+	isTerminal bool
+}
+
+type reformatLogger struct {
```

Comment:
Why do we need this?  We should instead just integrate with the https://godoc.org/github.com/go-kit/kit/log/term package instead of writing custom code.

Reply:
While if we wanted to just add color `go-kit/kit/log/term` would satisfy that need, this reformatter organizes how the data is displayed and color codes the keys.

For Example:
```go
	c := complex{
		A: "Awesome",
		B: 2,
	}
	errTermLogger.Log("key", "value", "err", errors.New("my bad"), logging.MessageKey(), "I am the term logger")
	warnTermLogger.Log(logging.MessageKey(), "Cool logger", "duration", time.Since(beginEleusinian))
	infoTermLogger.Log("key", c, "err", errors.New("my bad"), logging.MessageKey(), "I am the term logger")
	fmt.Println("---")
	errorLog.Log("key", "value", "err", errors.New("my bad"), logging.MessageKey(), "I am the reformat logger")
	warnLog.Log(logging.MessageKey(), "Cool logger", "duration", time.Since(beginEleusinian))
	infoLog.Log("key", c, "err", errors.New("my bad"), logging.MessageKey(), "I am the reformat logger")
```
Will Produce:
![image](https://user-images.githubusercontent.com/5015672/44550161-42e12c00-a6d8-11e8-939f-8fb4ddc9d567.png)